src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -39,7 +39,7 @@ internal async Task<ManagedIdentityResponse> SendTokenRequestForManagedIdentityA
         }
 
         // This method tries to create managed identity source for different sources, if none is created then defaults to IMDS.
-        private static async Task<AbstractManagedIdentity> SelectManagedIdentitySourceAsync(RequestContext requestContext)
+        private async Task<AbstractManagedIdentity> SelectManagedIdentitySourceAsync(RequestContext requestContext)
```

Comment:
This private method has no state. So it makes sense for it to be static. Is there a benefit for making this an instance method? Maybe `requestContext` should be part of the state?

Reply:
I think the performance gain in keeping it static is negligible for this scenario.